backend/src/main.rs

```diff
@@ -1,3 +1,24 @@
-fn main() {
-    println!("Hello, world!");
+use std::io;
+
+use axum::{http::HeaderValue, routing::get};
+use backend::{api, get_openapi_json, Api};
+use hyper::header::CONTENT_TYPE;
+use tower_http::cors::CorsLayer;
+use utoipa::OpenApi;
+use utoipa_scalar::{Scalar, Servable};
+
+#[tokio::main]
+async fn main() -> Result<(), io::Error> {
+    let app = axum::Router::new()
+        .layer(
+            CorsLayer::new()
+                .allow_origin("http://localhost:3000".parse::<HeaderValue>().unwrap())
+                .allow_headers([CONTENT_TYPE]),
+        )
+        .merge(Scalar::with_url("/api/docs", Api::openapi()))
+        .route("/api/openapi.json", get(get_openapi_json))
+        .nest("/api", api::router());
+
+    let listner = tokio::net::TcpListener::bind("0.0.0.0:5000").await?;
```

Comment:
Macだと5000番ポートがAirPlayのポートに割り振られていたり、Docker Registryで利用するポートが5000番だったりと競合しやすいので、5001番に変えた方がトラブルが減っていいかもしれません。（あくまでも個人の意見なので気にしなくても大丈夫です）
```suggestion
    let listner = tokio::net::TcpListener::bind("0.0.0.0:5001").await?;
```

Reply:
https://github.com/tus-project-team/program-grading-system/pull/38/commits/25215e8fe9a6caf3641095f1ce9fd5ac02d4a8a2 にて修正しました